### PR TITLE
191 develop determinism integration test

### LINKED ISSUE #191: develop determinism integration test
**Is your feature request related to a problem? Please describe.**
In order to provide stable and robust training, need to support **determinism training** and test it.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #191 .

### Description
This PR implemented a determinism integration test and fixed several small bugs.

### Status
**Ready**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### CODE REVIEW

File: tests/integration_determinism.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        in_channels=1,
+        out_channels=1,
+        channels=(4, 8, 16, 32),
+        strides=(2, 2, 2),
+        num_res_units=2,
+    ).to(device)
+
+    loss = DiceLoss(do_sigmoid=True)
+    opt = torch.optim.Adam(net.parameters(), 1e-4)
+    src = DataLoader(_TestBatch(), batch_size=batch_size)
```
Comment (wyli) [inline tests/integration_determinism.py]: the composed transforms in the spleen segmentation demo work well, i think we should put the transforms in this test so that we can always replicate the results

File: tests/integration_determinism.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        output = net(img.to(device))
+        step_loss = loss(output, seg.to(device))
+        step_loss.backward()
+        opt.step()
+        epoch_loss += step_loss.item()
+    epoch_loss /= step
+
+    print('Loss:', epoch_loss)
+    delta = abs(epoch_loss - 0.70605)
+    if delta > 0.0001:
```
Comment (wyli) [inline tests/integration_determinism.py]: could use `np.allclose` and this `run_test ` could return a bool

File: tests/integration_determinism.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+import numpy as np
+import torch
+from torch.utils.data import DataLoader, Dataset
+
+from monai.data.synthetic import create_test_image_2d
+from monai.losses.dice import DiceLoss
+from monai.networks.nets.unet import UNet
+
+
+def run_test(batch_size=64, train_steps=100, device=torch.device("cuda:0")):
```
Comment (wyli) [inline tests/integration_determinism.py]: by default batch 64 training_steps 100 means 2 iterations, maybe we can have a few more iterations for testing purposes?

File: tests/integration_determinism.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    class _TestBatch(Dataset):
+
+        def __getitem__(self, _unused_id):
+            im, seg = create_test_image_2d(128, 128, noise_max=1, num_objs=4, num_seg_classes=1)
+            return im[None], seg[None].astype(np.float32)
+
+        def __len__(self):
+            return train_steps
+
+    net = UNet(
```
Comment (wyli) [inline tests/integration_determinism.py]: I remember the densenet pipeline also works well in the demo, how could we make it reproducible? but maybe that's a network testing task

File: tests/integration_determinism.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        dimensions=2,
+        in_channels=1,
+        out_channels=1,
+        channels=(4, 8, 16, 32),
+        strides=(2, 2, 2),
+        num_res_units=2,
+    ).to(device)
+
+    loss = DiceLoss(do_sigmoid=True)
+    opt = torch.optim.Adam(net.parameters(), 1e-4)
```
Comment (wyli) [inline tests/integration_determinism.py]: just a comment that I tried learning rate = 1e-2 the loss decreases quickly

File: tests/integration_determinism.py (hunk 1 above)
Comment (Nic-Ma) [reply tests/integration_determinism.py]: Added transform in the latest patch.
Thanks.

File: tests/integration_determinism.py (hunk 2 above)
Comment (Nic-Ma) [reply tests/integration_determinism.py]: updated in the latest patch.
Thanks.

File: tests/integration_determinism.py (hunk 3 above)
Comment (Nic-Ma) [reply tests/integration_determinism.py]: As offline discussion, no need to change.

File: tests/integration_determinism.py (hunk 4 above)
Comment (Nic-Ma) [reply tests/integration_determinism.py]: all reasonable experiments should be another integration task.

File: tests/integration_determinism.py (hunk 5 above)
Comment (Nic-Ma) [reply tests/integration_determinism.py]: updated in the latest patch.
Thanks.

**Comment by Nic-Ma** [comment]
@wyli thanks for your review.
I already updated the code according to the comments.
Could you please check it again?
Thanks.
